core/vibes/soul/form/select/index.tsx

```diff
@@ -16,7 +16,9 @@ type Props = {
   label?: string;
   hideLabel?: boolean;
   variant?: 'round' | 'rectangle';
-  options: Array<{ label: string; value: string }>;
+  options: Array<
+    { label: string; value: string } | { label: string; action: () => void | Promise<void> }
+  >;
```

Comment:
I don't think it's appropriate for the `Select` component to take a server action. Instead, a `<form>` _outside_ `Select` should be used to handle the action. If we use a `<form>` inside `Select` then it will result in nested forms in other places where we use `Select` within a form